cool-core/src/test/java/com/nus/cool/functionality/IcebergLoaderTest.java

```diff
@@ -0,0 +1,68 @@
+package com.nus.cool.functionality;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nus.cool.core.iceberg.query.IcebergQuery;
+import com.nus.cool.core.iceberg.result.AggregatorResult;
+import com.nus.cool.core.iceberg.result.BaseResult;
+import com.nus.cool.model.CoolModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class IcebergLoaderTest extends CsvLoaderTest {
+    static final Logger logger = LoggerFactory.getLogger(IcebergLoaderTest.class);
+
+    @BeforeTest
+    public void setUp() {
+        logger.info("Start UnitTest " + IcebergLoaderTest.class.getSimpleName());
+    }
+
+    @AfterTest
+    public void tearDown() {
+        logger.info(String.format("Tear down UnitTest %s\n", IcebergLoaderTest.class.getSimpleName()));
+    }
+
+    @Test(dataProvider = "IceBergUnitTestDP", dependsOnMethods="CsvLoaderUnitTest")
+    public void IceBergTest(String datasetPath, String queryPath, List<BaseResult> out) throws Exception {
+        // load query
+        ObjectMapper mapper = new ObjectMapper();
+        IcebergQuery query = mapper.readValue(new File(queryPath), IcebergQuery.class);
+
+        // load .dz file
+        String dataSourceName = query.getDataSource();
+        CoolModel coolModel = new CoolModel(datasetPath);
+        coolModel.reload(dataSourceName);
+
+        // execute query
+        List<BaseResult> results = coolModel.olapEngine.performOlapQuery(coolModel.getCube(dataSourceName), query);
+
+        assert results.equals(out);
```

Comment:
Do not use `assert` keyword in java, use the `Assert.assert` from TestNG.
`assert` keyword is rarely used in Java. JVM will ignore `assert` command by default. 
In compiling, only you add flag `-eableassertions`  and `assert` is actived

Reply:
replaced